hpcgap/lib/vecmat.gd

```diff
@@ -431,6 +431,34 @@ DeclareOperation( "ImmutableMatrix",[IsObject,IsMatrix]);
 
 #############################################################################
 ##
+#F  ImmutableVector( <field>, <vector>[, <change>] ) . convert into "best" representation
+##
+##  <#GAPDoc Label="ImmutableVector">
+##  <ManSection>
+##  <Oper Name="ImmutableVector" Arg='field, vector[, change]'/>
+##
+##  <Description>
+##  returns an immutable vector equal to <A>vector</A> which is in the optimal
+##  (concerning space and runtime) representation for vectors defined over
+##  <A>field</A>. This means that vectors obtained by several calls of
+##  <Ref Oper="ImmutableVector"/> for the same <A>field</A> are compatible
+##  for fast arithmetic without need for field conversion.
+##  <P/>
+##  The input vector <A>vector</A> might change the representation,
+##  however the result of <Ref Oper="ImmutableVector"/> is not necessarily
+##  <E>identical</E> to <A>vector</A> if a conversion is not possible.
+##  <P/>
+##  If <A>change</A> is <K>true</K>, then <A>vector</A> may be changed to
+##  become immutable; otherwise it is copied first.
```

Comment:
(Indeed, I explicitly work around `CopyToVectorRep` potentially returning `fail` -- and this behavior of `CopyToVectorRep` makes it really annoying to use, and is the main reason why I am adding `ImmutableVector` instead of just porting `CopyToVectorRep` to plain GAP.